File: engine/model/src/main/scala/server/MapleShop.scala
package server

class MapleShop(private var _id: Int, private var _npcId: Int) {
  def id: Int = _id

  def npcId: Int = _npcId

  private var _items: Array[MapleShopItem] = new Array[MapleShopItem](0)

  def setItems(items: Array[MapleShopItem]):Unit = _items = items

  def items: Array[MapleShopItem] = _items

  def findBySlot(slot: Short): Option[MapleShopItem] = {
    if (slot < 0 || slot >= _items.length) {
      Option.empty
    }
    Option.apply(_items(slot))
  }

  def tokenValue: Int = 1000000000

  def token: Int = 4000313
}

File: engine/engine-model/src/main/java/server/MapleShop.java
package server;

import java.util.Optional;

public class MapleShop {
   private final Integer id;

   private final Integer npcId;

   private MapleShopItem[] items = new MapleShopItem[0];

   public MapleShop(Integer id, Integer npcId) {
      this.id = id;
      this.npcId = npcId;
   }

   public Optional<MapleShopItem> findBySlot(Short slot) {
      if (slot < 0 || slot >= items.length) {
         return Optional.empty();
      }
      return Optional.of(items[slot]);
   }

   public void setItems(MapleShopItem[] items) {
      this.items = items;
   }

   public MapleShopItem[] items() {
      return items;
   }

   public Integer id() {
      return id;
   }

   public Integer npcId() {
      return npcId;
   }

   public Integer tokenValue() {
      return 1000000000;
   }

   public Integer token() {
      return 4000313;
   }
}
